asv_bench/benchmarks/benchmarks.py

```diff
@@ -992,4 +993,106 @@ def time_float_32(self, shape, method):
+        self.index = IMPL.MultiIndex(
+            levels=[np.arange(10), np.arange(Nroot), np.arange(Nroot)],
+            codes=[
+                np.arange(10).repeat(N),
+                np.tile(np.arange(Nroot).repeat(Nroot), 10),
+                np.tile(np.tile(np.arange(Nroot), Nroot), 10),
+            ],
+        )
+        self.df1 = IMPL.DataFrame(
+            np.random.randn(len(self.index), col1), index=self.index
+        )
+        self.df2 = IMPL.DataFrame(np.random.randn(row2, col2))
+        execute(self.df1)
+        execute(self.df2)
+
+    def time_align_level(self, shapes):
+        left, right = self.df1.align(self.df2, level=1, copy=False)
+        execute(left)
+        execute(right)
+
+    # reindex returns  the same result as align. Approximately the same performance is expected.
+    def time_reindex_level(self, shapes):
+        execute(self.df2.reindex(self.index, level=1))
+
+
+class TimeDropDuplicatesDataframe:
+
+    params = [get_benchmark_shapes("TimeDropDuplicatesDataframe")]
+    param_names = ["shape"]
+
+    def setup(self, shape):
+        N = shape[0]
+        K = shape[1]
+        key1 = tm.makeStringIndex(N).values.repeat(K)
+        key2 = tm.makeStringIndex(N).values.repeat(K)
+        self.df = IMPL.DataFrame(
+            {"key1": key1, "key2": key2, "value": np.random.randn(N * K)}
+        )
+        self.df_nan = self.df.copy()
+        self.df_nan.iloc[:N, :] = np.nan
+        key1 = np.random.randint(0, K, size=N)
+        self.df_int = IMPL.DataFrame({"key1": key1})
+        execute(self.df), execute(self.df_nan)
+        execute(self.df_int)
+
+    def time_frame_drop_dups(self, shape):
+        execute(self.df.drop_duplicates(["key1", "key2"]))
+
+    def time_frame_drop_dups_inplace(self, shape):
+        self.df.drop_duplicates(["key1", "key2"], inplace=True)
+        execute(self.df)
+
+    def time_frame_drop_dups_na(self, shape):
+        execute(self.df_nan.drop_duplicates(["key1", "key2"]))
+
+    def time_frame_drop_dups_na_inplace(self, shape):
+        self.df_nan.drop_duplicates(["key1", "key2"], inplace=True)
+        execute(self.df_nan)
+
+    def time_frame_drop_dups_int(self, shape):
+        execute(self.df_int.drop_duplicates())
+
+    def time_frame_drop_dups_int_inplace(self, shape):
+        self.df_int.drop_duplicates(inplace=True)
+        execute(self.df_int)
+
+
+class TimeDropDuplicatesSeries:
+
+    params = [get_benchmark_shapes("TimeDropDuplicatesSeries")]
+    param_names = ["shape"]
+
+    def setup(self, shape):
+        N = shape[0]
+        self.s = IMPL.Series(np.random.randint(0, N // 10, size=N))
+        self.s_str = IMPL.Series(np.tile(tm.makeStringIndex(N // 10).values, 10))
+        execute(self.s), execute(self.s_str)
+
+    def time_series_drop_dups_int(self, shape):
+        execute(self.s.drop_duplicates())
+
+    def time_series_drop_dups_int_inplace(self, shape):
+        self.s.drop_duplicates(inplace=True)
+        execute(self.s)
```

Comment:
Modin does not have a specific implementation for different types for `drop_duplicates`. I propose to leave one type here. For example, the string type.

Reply:
Same for `TimeDropDuplicatesDataframe` class.

Reply:
Removed all drop_duplicates function except for strings